packages/wallets/ledger/src/adapter.ts

```diff
@@ -147,9 +147,15 @@ export class LedgerWalletAdapter extends BaseSignerWalletAdapter {
         try {
             try {
                 const transport = this._transport;
-                if (!transport) throw new WalletNotConnectedError();
+                const publicKey = this._publicKey;
+                if (!transport || !publicKey) throw new WalletNotConnectedError();
+
+                const appConfig = await getAppConfiguration(transport);
+                if (appConfig.version < '1.8.0') throw new WalletSignMessageError('Signing off-chain messages requires Solana Ledger App 1.8.0 or later');
+
+                const offchainMessage = new OffchainMessage({ message: Buffer.from(message.buffer), signerAddress: publicKey });
+                if (!offchainMessage.isLedgerSupported(appConfig.blindSigningEnabled)) throw new WalletSignMessageError('Ledger does not support signing this message. Either the message body is not printable ASCII and blind signing needs to be enabled, or the message is too long to be signed on Ledger.');
```

Comment:
Is it not possible to determine which of these two conditions is causing the inability to sign, rather than mentioning both?

Reply:
Addressed here: https://github.com/anza-xyz/wallet-adapter/pull/1094/commits/d5e1411d9c4da88ba885513b12e2ba1f6398f189